src/main/java/ch/uzh/ifi/hase/soprafs22/controller/ChatController.java

```diff
@@ -36,26 +40,28 @@ public class ChatController {
   public ResponseEntity<List<ChatOverViewGetDTO>> getFirstMessageOfEveryChat(@RequestHeader(value = "authorization", required = false) String token,
                                                                              @PathVariable(value = "userId") long userId) {
     userService.checkSpecificAccess(token, userId); // 404, 409
-    User user = new User();
-    user.setId(5L);
-    user.setGender(Gender.MALE);
-    user.setName("David");
-
-    User self = userService.getUserById(userId);
-    Message msg = new Message();
-    msg.setId(6L);
-    msg.setMessageType(MessageType.PLAIN_TEXT);
-    msg.setRead(false);
-    msg.setFromUserId(user.getId());
-    msg.setToUserId(self.getId());
-    msg.setContent("Hi, how are you");
-
-    ChatOverViewGetDTO chatOverViewGetDTO = new ChatOverViewGetDTO();
-    chatOverViewGetDTO.setUser(DTOMapper.INSTANCE.convertEntityToUserGetDTO(user));
-    chatOverViewGetDTO.setMessage(msg);
+
+      // get User by ID
+    User user = userService.getUserById(userId);
+    // get all matches from user
+    List<Match> matches = matchService.getMatches(user);
+    // get matchedUser
+    List<User> usersMatched = matchService.getUsersFromMatches(user, matches);
+    // get the last Message from the matches/ chats
+    List<Message> msg = chatService.getFirstMessages(matches);
 
     List<ChatOverViewGetDTO> chatOverViewGetDTOList = new ArrayList<>();
-    chatOverViewGetDTOList.add(chatOverViewGetDTO);
+
+      // mapp the messages, go through both lists and add them to chatOverViewGetDTOList
+      Iterator<User> matchedUser = usersMatched.iterator();
+      Iterator<Message> message = msg.iterator();
+
+      while (matchedUser.hasNext() && message.hasNext()) {
+          ChatOverViewGetDTO chatOverViewGetDTO = new ChatOverViewGetDTO();
+          chatOverViewGetDTO.setUser(DTOMapper.INSTANCE.convertEntityToUserGetDTO(matchedUser.next()));
```

Comment:
.next() returns the current object? (Question)

Reply:
yes